packages/steiger-plugin-fsd/src/no-segments-on-sliced-layers/index.ts

```diff
@@ -0,0 +1,34 @@
+import { sep } from 'node:path'
+
+import { getLayers, isSliced, conventionalSegmentNames } from '@feature-sliced/filesystem'
+import type { Diagnostic, Rule } from '@steiger/types'
+
+const noSegmentsOnSlicedLayers = {
+  name: 'no-segments-on-sliced-layers',
+  check(root) {
+    const diagnostics: Array<Diagnostic> = []
+    const layers = Object.values(getLayers(root))
+
+    for (const layer of layers) {
+      if (isSliced(layer)) {
+        for (const directChild of layer.children) {
+          if (directChild.type === 'folder') {
+            const folderName = directChild.path.split(sep).pop()
```

Comment:
suggestion (non-blocking): you can use `basename` from `node:path` to get the folder name:
```suggestion
            const folderName = basename(directChild.path)
```

Reply:
👍
Yes, it looks more concise, graceful and does not reinvent the wheel